src/claimant.js

```diff
@@ -0,0 +1,171 @@
+import xdr from './generated/stellar-xdr_generated';
+import { Keypair } from './keypair';
+import { StrKey } from './strkey';
+
+/**
+ * Claimant class represents an xdr.Claimant
+ *
+ * The claim predicate is optional, it defaults to unconditional if none is specified.
+ *
+ * @constructor
+ * @param {string} destination - The destination account ID.
+ * @param {xdr.ClaimPredicate} [predicate] - The claim predicate.
+ */
+export class Claimant {
+  constructor(destination, predicate) {
+    if (destination && !StrKey.isValidEd25519PublicKey(destination)) {
+      throw new Error('Destination is invalid');
+    }
+    this._destination = destination;
+
+    if (!predicate) {
+      this._predicate = xdr.ClaimPredicate.claimPredicateUnconditional();
+    } else if (predicate instanceof xdr.ClaimPredicate) {
+      this._predicate = predicate;
+    } else {
+      throw new Error('Predicate should be an xdr.ClaimPredicate');
+    }
+  }
+
+  /**
+   * Returns an unconditional claim predicate
+   * @Return {xdr.ClaimPredicate}
+   */
+  static claimPredicateUnconditional() {
+    return xdr.ClaimPredicate.claimPredicateUnconditional();
+  }
+
+  /**
+   * Returns an `and` claim predicate
+   * @param {xdr.ClaimPredicate} left an xdr.ClaimPredicate
+   * @param {xdr.ClaimPredicate} right an xdr.ClaimPredicate
+   * @Return {xdr.ClaimPredicate}
+   */
```

Comment:
This is more of a general question since I've seen these kinds of doc-strings in a lot of other places: _why do we have them?_ They don't really provide meaningful information that isn't apparent from the function signature. It makes some sense in JS since there are no types, but I've seen it in the Go SDK as well and it feels like  clutter. Something like 
```js
/// @Return a new claim predicate that is the logical AND of both parameters.
```
could convey the same idea and look cleaner

...or maybe I'm just tired of reading `ClaimClaimableBalance/ClaimPredicate` over and over as I work on integration tests :joy:

Reply:
this is JSDOC format which is used to automatically generate the docs :) https://jsdoc.app/